internal/model/rwfolder.go

```diff
@@ -1222,8 +1223,28 @@ func (p *rwFolder) performFinish(state *sharedPullerState) error {
 		}
 	}
 
-	// Set the correct timestamp on the new file
+	// Verify if a file will be overwritten and if we are up to date about its current mtime
 	t := time.Unix(state.file.Modified, 0)
+	existingInfo, existingErr := os.Lstat(state.realName)
+	if existingErr == nil {
+		p.model.fmut.RLock()
+		mtimeRepo, _ := p.model.folderMtimeRepos[p.folder]
+		p.model.fmut.RUnlock()
+		mtime := existingInfo.ModTime()
+		if mtimeRepo != nil {
+			mtime = mtimeRepo.GetMtime(state.file.Name, mtime)
+			// TODO should break here. If we need the mtime value from the mtimeRepo we are 'up to date' anyway
+		}
+		if state.modified != existingInfo.ModTime().Unix() {
+			// File changed and we didn't know about it, make a conflict
+			if debug {
+				l.Debugf("%v performFinish %s: %q; outdated: %v %v", p, p.folder, state.file.Name, state.modified, t.Unix())
+			}
+			// TODO do we need to scan here? See p.inConflict
```

Comment:
I don't think we do.
Can't we just make a copy of curFile, update the mtime and do:
```
if curFile.WinsConflict(state.targetFile) {
  magically rename temp file to target file + conflict suffix somehow
} else {
  rename curFile to conflict
  move temp file to place
}
```